### PR TITLE
Add documentation for Azure OpenAI

### PR DESCRIPTION
**Pull request by shivam-51**
Thank you for helping make the Dapr documentation better!

**Please follow this checklist before submitting:**
- [x] Commits are signed with Developer Certificate of Origin (DCO - [learn more](https://docs.dapr.io/contributing/contributing-overview/#developer-certificate-of-origin-signing-your-work))
- [x] [Read the contribution guide](https://docs.dapr.io/contributing/docs-contrib/contributing-docs/)
- [ ] Commands include options for Linux, MacOS, and Windows within codetabs
- [x] New file and folder names are globally unique
- [x] Page references use shortcodes instead of markdown or URL links
- [ ] Images use HTML style and have alternative text
- [x] Places where multiple code/command options are given have codetabs

In addition, please fill out the following to help reviewers understand this pull request:

## Description

Add documentation for Azure OpenAI output binding
Refer to this PR for more details https://github.com/dapr/components-contrib/pull/2965

## Issue reference

https://github.com/dapr/components-contrib/issues/2716

### CODE REVIEW

**Comment by shivam-51** [comment]
@ItalyPaleAle Please have a look.

File: daprdocs/content/en/reference/components-reference/supported-bindings/openai.md [hunk 1]
```diff
@@ -0,0 +1,236 @@
...
+title: "Azure OpenAI binding spec"
+linkTitle: "Azure OpenAI"
+description: "Detailed documentation on the Azure OpenAI binding component"
+aliases:
+  - "/operations/components/setup-bindings/supported-bindings/openai/"
+---
+
+## Component format
+
+To setup OpenAI binding create a component of type `bindings.azure.openai`. See [this guide]({{< ref "howto-bindings.md#1-create-a-binding" >}}) on how to create and apply a binding configuration.
```
Comment (ItalyPaleAle) [inline daprdocs/content/en/reference/components-reference/supported-bindings/openai.md]: It's important we always call this "Azure OpenAI"

```suggestion
To setup an Azure OpenAI binding create a component of type `bindings.azure.openai`. See [this guide]({{< ref "howto-bindings.md#1-create-a-binding" >}}) on how to create and apply a binding configuration.
```

File: daprdocs/content/en/reference/components-reference/supported-bindings/openai.md [hunk 2]
```diff
@@ -0,0 +1,236 @@
...
+See [this](https://learn.microsoft.com/en-us/azure/cognitive-services/openai/overview/) for the documentation for Azure OpenAI Service.
```
Comment (ItalyPaleAle) [inline daprdocs/content/en/reference/components-reference/supported-bindings/openai.md]:
```suggestion
See [this](https://learn.microsoft.com/azure/cognitive-services/openai/overview/) for the documentation for Azure OpenAI Service.
```

File: daprdocs/content/en/reference/components-reference/supported-bindings/openai.md [hunk 3]
```diff
@@ -0,0 +1,236 @@
...
+{{% alert title="Warning" color="warning" %}}
+The above example uses apiKey as plain strings. It is recommended to use a secret store for the secrets as described [here]({{< ref component-secrets.md >}}).
+{{% /alert %}}
+
+## Spec metadata fields
+
+| Field              | Required | Binding support | Details | Example |
+|--------------------|:--------:|--------|---------|---------|
+| `endpoint` | Y | Input/Output | Azure OpenAI service endpoint URL. | `"https://myopenai.openai.azure.com"` |
+| `apiKey` | Y* | Input/Output | The access key of the Azure OpenAI service. Only required when not using Azure AD authentication. | `"1234567890abcdef"` |
```
Comment (ItalyPaleAle) [inline daprdocs/content/en/reference/components-reference/supported-bindings/openai.md]:
```suggestion
| `endpoint` | Y | Output | Azure OpenAI service endpoint URL. | `"https://myopenai.openai.azure.com"` |
| `apiKey` | Y* | Output | The access key of the Azure OpenAI service. Only required when not using Azure AD authentication. | `"1234567890abcdef"` |
```

File: daprdocs/content/en/reference/components-reference/supported-bindings/openai.md [hunk 4]
```diff
@@ -0,0 +1,236 @@
...
+
+
+## Related links
+
+- [Basic schema for a Dapr component]({{< ref component-schema >}})
+- [Bindings building block]({{< ref bindings >}})
+- [How-To: Trigger application with input binding]({{< ref howto-triggers.md >}})
+- [How-To: Use bindings to interface with external resources]({{< ref howto-bindings.md >}})
+- [Bindings API reference]({{< ref bindings_api.md >}})
+- [Azure OpenAI Rest examples](https://learn.microsoft.com/en-us/azure/cognitive-services/openai/reference)
```
Comment (ItalyPaleAle) [inline daprdocs/content/en/reference/components-reference/supported-bindings/openai.md]:
```suggestion
- [Azure OpenAI Rest examples](https://learn.microsoft.com/azure/cognitive-services/openai/reference)
```

**Review by ItalyPaleAle — COMMENTED** [review]
Thanks for creating these docs!

File: daprdocs/content/en/reference/components-reference/supported-bindings/openai.md [hunk 5]
```diff
@@ -0,0 +1,236 @@
...
+```yaml
+apiVersion: dapr.io/v1alpha1
+kind: Component
+metadata:
+  name: <NAME>
+spec:
+  type: bindings.azure.openai
+  version: v1
+  metadata:
+  - name: apiKey
```
Comment (msfussell) [inline daprdocs/content/en/reference/components-reference/supported-bindings/openai.md]:
```suggestion
  - name: apiKey # Required
```

File: daprdocs/content/en/reference/components-reference/supported-bindings/openai.md [hunk 6]
```diff
@@ -0,0 +1,236 @@
...
+    value: "1234567890abcdef"
+  - name: endpoint # Required
+    value: "https://myopenai.openai.azure.com"
+  - name: deploymentId # Required
+    value: "my-model"
+```
+{{% alert title="Warning" color="warning" %}}
+The above example uses apiKey as plain strings. It is recommended to use a secret store for the secrets as described [here]({{< ref component-secrets.md >}}).
```
Comment (msfussell) [inline daprdocs/content/en/reference/components-reference/supported-bindings/openai.md]:
```suggestion
The above example uses `apiKey` as  a plain string. It is recommended to use a secret store for the secrets as described [here]({{< ref component-secrets.md >}}).
```

File: daprdocs/content/en/reference/components-reference/supported-bindings/openai.md [hunk 7]
```diff
@@ -0,0 +1,236 @@
...
+    "maxTokens":5
+    }
+}
+```
+
+The data parameters are:
+
+- `prompt` - string that specifies the prompt to generate completions for.
+- `maxTokens` - (optional) defines the max number of tokens to generate. Defaults to 16 for completion API.
+- `temperature` - (optional) defines the sampling temperature between 0 and 2. Higher values like 0.8 will make the output more random, while lower values like 0.2 will make it more focused and deterministic. Defaults to 1.0 for completion API.
```
Comment (msfussell) [inline daprdocs/content/en/reference/components-reference/supported-bindings/openai.md]:
```suggestion
- `temperature` - (optional) defines the sampling temperature between 0 and 2. Higher values like 0.8 make the output more random, while lower values like 0.2 make it more focused and deterministic. Defaults to 1.0 for completion API.
```

File: daprdocs/content/en/reference/components-reference/supported-bindings/openai.md [hunk 8]
```diff
@@ -0,0 +1,236 @@
...
+  curl -d '{ "data": {"prompt": "A dog is ", "maxTokens":15}, "operation": "completion" }' \
+        http://localhost:<dapr-port>/v1.0/bindings/<binding-name>
+  ```
+  {{% /codetab %}}
+
+{{< /tabs >}}
+
+#### Response
+
+The response body will contain the following JSON:
```
Comment (msfussell) [inline daprdocs/content/en/reference/components-reference/supported-bindings/openai.md]:
```suggestion
The response body contains the following JSON:
```

File: daprdocs/content/en/reference/components-reference/supported-bindings/openai.md [hunk 9]
```diff
@@ -0,0 +1,236 @@
...
+```
+
+The data parameters are:
+
+- `messages` - array of messages that will be used to generate chat completions.
+Each message is of the form:
+  - `role` - string that specifies the role of the message. Can be either `user`, `system` or `assistant`.
+  - `message` - string that specifies the conversation message for the role.
+- `maxTokens` - (optional) defines the max number of tokens to generate. Defaults to 16 for the chat completion API.
+- `temperature` - (optional) defines the sampling temperature between 0 and 2. Higher values like 0.8 will make the output more random, while lower values like 0.2 will make it more focused and deterministic. Defaults to 1.0 for the chat completion API.
```
Comment (msfussell) [inline daprdocs/content/en/reference/components-reference/supported-bindings/openai.md]:
```suggestion
- `temperature` - (optional) defines the sampling temperature between 0 and 2. Higher values like 0.8 make the output more random, while lower values like 0.2 make it more focused and deterministic. Defaults to 1.0 for the chat completion API.
```

File: daprdocs/content/en/reference/components-reference/supported-bindings/openai.md [hunk 10]
```diff
@@ -0,0 +1,236 @@
...
+}' \
+http://localhost:<dapr-port>/v1.0/bindings/<binding-name>
+  ```
+  {{% /codetab %}}
+
+{{< /tabs >}}
+
+#### Response
+
+The response body will contain the following JSON:
```
Comment (msfussell) [inline daprdocs/content/en/reference/components-reference/supported-bindings/openai.md]:
```suggestion
The response body contains the following JSON:
```

File: daprdocs/content/en/reference/components-reference/supported-bindings/openai.md [hunk 11]
```diff
@@ -0,0 +1,236 @@
...
+{{% /alert %}}
+
+## Spec metadata fields
+
+| Field              | Required | Binding support | Details | Example |
+|--------------------|:--------:|--------|---------|---------|
+| `endpoint` | Y | Output | Azure OpenAI service endpoint URL. | `"https://myopenai.openai.azure.com"` |
+| `apiKey` | Y* | Output | The access key of the Azure OpenAI service. Only required when not using Azure AD authentication. | `"1234567890abcdef"` |
+| `deploymentId` | Y | Output | The name of the model deployment. | `"my-model"` |
+
```
Comment (msfussell) [inline daprdocs/content/en/reference/components-reference/supported-bindings/openai.md:46]: Can you also include the Azure AAD authentication fields into this table, as shown in the example below.  You can take the same descriptions from here https://docs.dapr.io/reference/components-reference/supported-bindings/eventgrid/

**Review by msfussell — COMMENTED** [review]
Great documentation! Left a few comments

File: daprdocs/content/en/reference/components-reference/supported-bindings/openai.md (hunk 5 above)
Comment (shivam-51) [reply daprdocs/content/en/reference/components-reference/supported-bindings/openai.md]: Done.

File: daprdocs/content/en/reference/components-reference/supported-bindings/openai.md (hunk 11 above)
Comment (shivam-51) [reply daprdocs/content/en/reference/components-reference/supported-bindings/openai.md:46]: Done. Please have a look.

**Comment by shivam-51** [comment]
@msfussell Please have a look.

**Comment by shivam-51** [comment]
@mukundansundar Please have a look.

**Review by msfussell — APPROVED** [review]
LGTM

File: daprdocs/content/en/reference/components-reference/supported-bindings/openai.md [hunk 12]
```diff
@@ -0,0 +1,239 @@
...
+- `temperature` - (optional) defines the sampling temperature between 0 and 2. Higher values like 0.8 make the output more random, while lower values like 0.2 make it more focused and deterministic. Defaults to 1.0 for completion API.
+- `topP` - (optional) defines the sampling temperature. Defaults to 1.0 for completion API.
+- `n` - (optional) defines the number of completions to generate. Defaults to 1 for completion API.
+- `presencePenalty` - (optional) Number between -2.0 and 2.0. Positive values penalize new tokens based on whether they appear in the text so far, increasing the model's likelihood to talk about new topics. Defaults to 0.0 for completion API.
+- `frequencyPenalty` - (optional) Number between -2.0 and 2.0. Positive values penalize new tokens based on their existing frequency in the text so far, decreasing the model's likelihood to repeat the same line verbatim. Defaults to 0.0 for completion API.
+
+Read more about the importance and usage of these parameters in the [Azure OpenAI API documentation](https://learn.microsoft.com/en-us/azure/ai-services/openai/reference).
```
Comment (hhunter-ms) [inline daprdocs/content/en/reference/components-reference/supported-bindings/openai.md]:
```suggestion
Read more about the importance and usage of these parameters in the [Azure OpenAI API documentation](https://learn.microsoft.com/azure/ai-services/openai/reference).
```

File: daprdocs/content/en/reference/components-reference/supported-bindings/openai.md [hunk 13]
```diff
@@ -0,0 +1,239 @@
...
+
+
+## Related links
+
+- [Basic schema for a Dapr component]({{< ref component-schema >}})
+- [Bindings building block]({{< ref bindings >}})
+- [How-To: Trigger application with input binding]({{< ref howto-triggers.md >}})
+- [How-To: Use bindings to interface with external resources]({{< ref howto-bindings.md >}})
+- [Bindings API reference]({{< ref bindings_api.md >}})
+- [Azure OpenAI Rest examples](https://learn.microsoft.com/en-us/azure/ai-services/openai/reference)
```
Comment (hhunter-ms) [inline daprdocs/content/en/reference/components-reference/supported-bindings/openai.md]:
```suggestion
- [Azure OpenAI Rest examples](https://learn.microsoft.com/azure/ai-services/openai/reference)
```

**Pull request merged**